src/Appwrite/Database/Adapter/MySQL.php

```diff
@@ -568,8 +569,15 @@ public function getCollection(array $options)
 
             //$path = implode('.', $path);
 
+            $castToInt = array_key_exists($key, $filterTypes) && $filterTypes[$key] === 'numeric';
+
             $key = $this->getPDO()->quote($key, PDO::PARAM_STR);
             $value = $this->getPDO()->quote($value, PDO::PARAM_STR);
+
+            if ($castToInt) {
```

Comment:
Wouldn't `(int)` be good enough for casting the value?

Reply:
Then I needed to skip calling `pdo->quote`. I did this, but with `(float)` now - to support decimals.